File: Samples/Evaluate/how-to-replace-standard-spreadsheetcontrol-command-with-your-own-custom-command-t163272-14.1.7-/VB/SpreadsheetCustomCommand/CustomFormatClearContentsCommand.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraSpreadsheet
Imports DevExpress.XtraSpreadsheet.Commands

Namespace SpreadsheetCustomCommand
    #Region "#customcommand"
    Public Class CustomFormatClearContentsCommand
        Inherits FormatClearContentsCommand

        Public Sub New(ByVal control As ISpreadsheetControl)
            MyBase.New(control)
        End Sub

        Protected Overrides Sub ExecuteCore()
            MessageBox.Show("Custom command executed")
            MyBase.ExecuteCore()
        End Sub
    End Class
    #End Region ' #customcommand
End Namespace

File: Samples/Evaluate/how-to-replace-standard-spreadsheetcontrol-command-with-your-own-custom-command-t163272-14.1.7-/CS/SpreadsheetCustomCommand/CustomFormatClearContentsCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraSpreadsheet;
using DevExpress.XtraSpreadsheet.Commands;

namespace SpreadsheetCustomCommand
{
    #region #customcommand
    public class CustomFormatClearContentsCommand : FormatClearContentsCommand {
        public CustomFormatClearContentsCommand(ISpreadsheetControl control)
            : base(control) {
        }

        protected override void ExecuteCore() {
            MessageBox.Show("Custom command executed");
            base.ExecuteCore();
        }
    }
    #endregion #customcommand
}
